packages/browser/src/login/oidc/redirectHandler/AuthCodeRedirectHandler.ts

```diff
@@ -146,44 +147,44 @@ export class AuthCodeRedirectHandler implements IRedirectHandler {
         errorIfNull: true,
       }
     )) as string;
-    const isDpop =
-      (await this.storageUtility.getForUser(storedSessionId, "dpop")) ===
-      "true";
 
-    // Since we throw if not found, the type assertion is okay
-    const issuer = (await this.storageUtility.getForUser(
+    const {
+      issuerConfig,
+      codeVerifier,
+      redirectUrl: storedRedirectIri,
+      dpop: isDpop,
+    } = await loadOidcContextFromStorage(
       storedSessionId,
-      "issuer",
-      { errorIfNull: true }
-    )) as string;
+      this.storageUtility,
+      this.issuerConfigFetcher
+    );
 
-    // Store the current session ID specifically in 'localStorage' (i.e., not using
-    // any other storage mechanism), as we don't deem this information to be
-    // sensitive, and we want to ensure it survives a browser tab refresh.
-    window.localStorage.setItem(KEY_CURRENT_SESSION, storedSessionId);
+    if (codeVerifier === undefined) {
+      throw new Error(
+        `The code verifier for session ${storedSessionId} is missing from storage.`
+      );
+    }
+
+    if (storedRedirectIri === undefined) {
+      throw new Error(
+        `The redirect URL for session ${storedSessionId} is missing from storage.`
+      );
+    }
 
-    const issuerConfig = await this.issuerConfigFetcher.fetchConfig(issuer);
     const client: IClient = await this.clientRegistrar.getClient(
       { sessionId: storedSessionId },
       issuerConfig
```

Comment:
Can we absolutely trust the `issuerConfig` in local storage not to have been tampered with / could this open a security attack vector? (tamper with that value, user gets taken somewhere that's not their OIDC provider, or something similar?

Reply:
Technically no, we cannot trust blindly the values in local storage. What we could do (I believe we talked about this recently) is to hash the session information in storage before the redirect, send it in the `state`, and use it to verify non-tampering when redirected back. I'll create a dedicated ticket.